tests/core/util/Jws.spec.ts

```diff
@@ -0,0 +1,59 @@
+import Cryptography from '../../../lib/core/versions/latest/util/Cryptography';
+import ErrorCode from '../../../lib/core/versions/latest/ErrorCode';
+import Jws from '../../../lib/core/versions/latest/util/Jws';
+import KeyUsage from '../../../lib/core/versions/latest/KeyUsage';
+import SidetreeError from '../../../lib/core/SidetreeError';
+
+describe('Jws', async () => {
+  describe('parse()', async () => {
+    it('should throw error if protected header contains unexpected property.', async () => {
+      const signingKeyId = '#signingKey';
+      const [, signingPrivateKey] = await Cryptography.generateKeyPairHex(signingKeyId, KeyUsage.signing);
+
+      const protectedHeader = {
+        unknownProperty: 'anyValue',
+        kid: signingKeyId,
+        alg: 'ES256K'
+      };
+
+      const payload = { anyProperty: 'anyValue' };
+
+      const jws = await Jws.sign(protectedHeader, payload, signingPrivateKey);
+
+      expect(() => { Jws.parse(jws); }).toThrow(new SidetreeError(ErrorCode.JwsProtectedHeaderMissingOrUnknownProperty));
+    });
+
+    it('should throw error if `kid` in header is missing or is in incorrect type.', async () => {
+      const signingKeyId = '#signingKey';
+      const [, signingPrivateKey] = await Cryptography.generateKeyPairHex(signingKeyId, KeyUsage.signing);
+
+      const protectedHeader = {
+        kid: true, // Incorect type.
+        alg: 'ES256K'
+      };
+
+      const payload = { anyProperty: 'anyValue' };
+
+      const jws = await Jws.sign(protectedHeader, payload, signingPrivateKey);
+
+      expect(() => { Jws.parse(jws); }).toThrow(new SidetreeError(ErrorCode.JwsProtectedHeaderMissingOrIncorrectKid));
+    });
+
+    it('should throw error if `alg` in header is missing or is in incorrect type.', async () => {
+      const signingKeyId = '#signingKey';
+      const [, signingPrivateKey] = await Cryptography.generateKeyPairHex(signingKeyId, KeyUsage.signing);
+
+      const protectedHeader = {
+        kid: signingKeyId,
+        alg: true // Incorect type.
+      };
+
+      const payload = { anyProperty: 'anyValue' };
+
+      const jws = await Jws.sign(protectedHeader, payload, signingPrivateKey);
```

Comment:
We should probably have a test for when payload is a string if we want to support it.

Reply:
Added a note saying that this is mainly created for tests to create valid data, and the path is hit many times already.